app/forms/post_form.rb

```diff
@@ -0,0 +1,56 @@
+class PostForm
+  include ActiveModel::Model
+  include Virtus.model
+  extend CarrierWave::Mount
+
+  validates :title,   presence: true, length: { maximum: 50 }
+  validates :content, presence: true, length: { maximum: 10000 }
+
+  attribute :user_id,     Integer
+  attribute :category_id, Integer
+  attribute :title,       String
+  attribute :content,     String
+  attribute :status,      Boolean
+
+  mount_uploader :picture, PictureUploader
+
+  attr_accessor :pictures
+
+  def initialize(post = Post.new)
+    @post = post
+    self.attributes = @post.attributes if @post.persisted?
+  end
+
+  def assign_attributes(params = {})
+    @params = params
+    # 画像情報からインスタンスを生成する
+    attributes = params.to_h[:pictures_attributes]
+    @pictures ||= []
+    attributes&.map do |attribute|
+      picture = Picture.new(attribute)
+      @pictures.push(picture)
+    end
+    # 画像以外の情報でpostを更新する
+    @params.delete(:pictures_attributes)
+    @post.assign_attributes(params) if @post.persisted?
+    super(params)
```

Comment:
paramsはattr_accessorで定義していないから、`@`付けないと参照できなさそうだけど、これで参照できてます？
```suggestion
    @post.assign_attributes(@params) if @post.persisted?
    super(@params)
```

Reply:
`@`をつけてもつけなくても動いたのですが付けて修正しました。
私の認識としては、異なるメソッドで呼び出す際はつける必要があると思っていたのですが、
どういった場合につけるべきなのでしょうか。

Reply:
> @をつけてもつけなくても動いたのですが付けて修正しました。

`params`は引数で受け取ってるのでローカル変数でしたね。。失礼しました。

> 私の認識としては、異なるメソッドで呼び出す際はつける必要があると思っていたのですが

`@`マークつけるとインスタンス変数になり、認識の通り、異なるメソッド間でも使えます。
attr_accessorを定義すると`@マーク`付けなくても呼び出せるようになりますが、これは内部的にrailsがゲッターとセッターを作ってくれるからです。

`attr_accessor :hoge`ってやると、こんな感じのメソッドを作ってくれる。
```
def hoge
  @hoge
end

def hoge=(value)
  @hoge = value
end

```

Reply:
> attr_accessorを定義すると@マーク付けなくても呼び出せるようになりますが、これは内部的にrailsがゲッターとセッターを作ってくれるからです。

内部的な挙動は確認していませんでした。勉強になりました！